foundations/javascript_basics/fundamentals-5.md

```diff
@@ -28,47 +28,116 @@ As an example of what we mean, let's consider a `sumOfTripledEvens` function. It
+Let's say we had a function, `isOdd` that returns either `true` if a number is odd or `false` if it isn't. 
+
+The `.filter(callback)` expects the `callback` to return either `true` or `false`. If it returns `true`, the value is included in the output. Otherwise, it isn't.
+Consider the array from our previous example, `[1, 2, 3, 4, 5]`. 
+If we wanted to remove all even numbers from this array, we could use `.filter()` like this:
+
+```js
+arr.filter(isOdd); // Outputs [1, 3, 5];
+```
+
+- `.filter(isOdd)` will iterate through the `arr` and pass **every value** into the `isOdd()` callback. 
+- `.isOdd()` will return `true` when the value is odd, which means this value is included in the output.
+- If it's an even number, `isOdd()` will return `false` and not include it in the final output.
+
+#### The .reduce() method
+
+Finally, let's say that we wanted to multiply all of the numbers in our `arr` together like this: `1 * 2 * 3 * 4 * 5`.
+First, we'd have to declare a variable `total` and initialize it to 0. Then, we'd iterate through the array with a `for` loop and multiply the `total` by the current number.
+
+But we don't actually need to do all of that, we have our `.reduce(callback, initialValue)` method that will do the job for us. Just like `.map()` and `.reduce()` it expects a callback function.
+However, it also takes in an `initialValue` which makes our lives easier and we don't have to declare a variable like `const total = 1` and instead `.reduce()` does that for us.
+
+```js
+arr.reduce(((total, currentItem) => total * currentItem), 1); // Outputs [1, 2, 6, 24, 120];
+```
+
+In the above function, we: 
+- Pass in a callback function, which is `(total, currentItem) => total * currentItem`.
+- Initialise total to `1` in the second argument.
+
+So what `.reduce()` will do, is it will once again go through every element in `arr` and apply the `callback` function to it. It then **stores** the output in `total`, which is 1 at the beginning. 
+
+#### Summary 
+
+You've learnt about the three powerful array methods which are `.map()`, `.filter()` and `.reduce()`. They allow us to write shorter code that is more readable and less prone to bugs. 
 
 For a quick recap of these array methods, consider this picture which should visually explain them in terms of sandwiches:
 
 ![Alt text](https://static.observableusercontent.com/thumbnail/bea194824f0d5842addcb7910bb488795c6f80f143ab5332b28a317ebcecd603.jpg)
 
-Don't worry if you don't fully understand these methods at this point, the above example is just a teaser to demonstrate their power. The assignment material provided will go deeper into the specifics of these functions.
+Your task is now to rewrite the `sumOfTripledEvens(array)` function using these three methods. **Hint**: you don't need to declare any variables, just having a single `return` within the function will be enough.
```

Comment:
```suggestion
Let's do some quick practice before your assignment! Rewrite the `sumOfTripledEvens(array)` function using these three methods.
```
I don't think that hint is particularly helpful, they will see once they click the result.